src/alert-manager/src/alert-handler/controllers/mail.js

```diff
@@ -0,0 +1,167 @@
+    if (jobName) {
+      promises.push(
+        new Promise(function (resolve) {
+          return unirest
+            .get(`${url}/api/v2/jobs/${jobName}`)
+            .headers({
+              Authorization: `Bearer ${token}`,
+              'Content-Type': 'application/json',
+            })
+            .end(function (res) {
+              resolve([res.body.jobStatus.username, alert]);
+            });
+        }),
+      );
+    }
+  });
+
+  const alertsGrouped = {};
+  return Promise.all(promises)
+    .then(function (values) {
+      values.forEach(function (value) {
+        const username = value[0];
+        const alert = value[1];
+        if (username in alertsGrouped) {
+          alertsGrouped[username].push(alert);
+        } else {
+          alertsGrouped[username] = [alert];
+        }
+      });
+      return alertsGrouped;
+    })
+    .catch(function (data) {
+      console.error(data);
+    });
+};
+
+const getUserEmail = (username, url, token) => {
+  return new Promise(function (resolve) {
+    return unirest
+      .get(`${url}/api/v2/users/${username}`)
+      .headers({
+        Authorization: `Bearer ${token}`,
+        'Content-Type': 'application/json',
+      })
+      .end(function (res) {
+        resolve(res.body.email);
+      });
+  });
+};
+
+const sendEmail = async (req, res) => {
+  // send email to admin
+  email
+    .send({
+      template: 'general-templates',
+      message: {
+        to: process.env.EMAIL_CONFIGS_RECEIVER,
+      },
+      locals: {
+        cluster_id: process.env.CLUSTER_ID,
+        alerts: req.body.alerts,
+        groupLabels: req.body.groupLabels,
+        externalURL: req.body.externalURL,
+      },
+    })
+    .then(function (res) {
+      console.log(
+        `alert-handler successfully send email to admin at ${process.env.EMAIL_CONFIGS_RECEIVER}`,
+      );
+    })
+    .catch(console.error);
+
+  // send email to job user when possible
+  // group alerts by username
+  const url = process.env.REST_SERVER_URI;
+  const token = req.token;
+
+  const alertsGrouped = await getAlertsGroupedByUser(
+    req.body.alerts,
+    url,
+    token,
+  );
+
+  if (alertsGrouped) {
+    // send emails to different users separately
+    Object.keys(alertsGrouped).forEach(async (username) => {
+      const userEmail = await getUserEmail(username, url, token);
+      email
+        .send({
+          template: 'general-templates',
+          message: {
+            to: userEmail,
+          },
+          locals: {
+            cluster_id: process.env.CLUSTER_ID,
+            alerts: alertsGrouped[username],
+            groupLabels: req.body.groupLabels,
+            externalURL: req.body.externalURL,
+          },
+        })
+        .then(function (res) {
+          console.log(
+            `alert-handler successfully send email to ${username} at ${userEmail}`,
+          );
+        })
+        .catch(console.error);
+    });
+  }
+
+  res.status(200).json({
```

Comment:
DO NOT return 200, return `send-email` response, let alert-manager decided if needs retry

Reply:
> DO NOT return 200, return `send-email` response, let alert-manager decided if needs retry

fixed